File: python/sieve/sieve.py
def primes(limit):
    result = []
    primes = [True if x >= 2 else False for x in range(limit + 1)]

    for i, prime in enumerate(primes):
        if not prime:
            continue

        result.append(i)

        for j in range(i, limit + 1, i):
            primes[j] = False

    return result


if __name__ == "__main__":
    print(f"{primes(100)=}")

File: go/sieve/sieve.go
package sieve

// Sieve gives all primes up to `limit`
func Sieve(limit int) []int {
	primes := []int{}
	p := map[int]bool{}

	for i := 2; i <= limit; i++ {
		p[i] = true
	}

	for i := 2; i <= limit; i++ {
		if !p[i] {
			continue
		}
		primes = append(primes, i)
		for j := i; j <= limit; j += i {
			p[j] = false
		}
	}
	return primes
}
